xbmc/network/TCPServer.cpp

```diff
@@ -227,6 +227,9 @@ void CTCPServer::Announce(ANNOUNCEMENT::AnnouncementFlag flag,
                           const std::string& message,
                           const CVariant& data)
 {
+  if (!m_connections.size())
```

Comment:
Is this threadsafe? A bit below while iterating m_connections Kodi obtains the lock

Reply:
but is within the loop the size is read before,
to make things as before i could put the size value in to a new variable and use it also in the loop

Reply:
Disregard, lock is in the actual connection not on the vector